### PR TITLE
terminate the agent active call

### PR DESCRIPTION
**Pull request by dvillarama**
brad,  this is the zestphone part.

### CODE REVIEW

File: app/controllers/telephony/agents_controller.rb [hunk 1]
```diff
@@ -7,6 +7,17 @@ def show_by_csr_id
       render json: agent.as_json(only: [:id, :csr_id, :name], methods: :active_conversation_id)
     end
 
+    def terminate_active_call
+      agent = Agent.find_by_csr_id params[:id]
+
+      if agent
+        agent.terminate_active_call
```
Comment (bradleyjucsc) [inline app/controllers/telephony/agents_controller.rb]: Please wrap this line in
`agent.with_lock do`
  # Your line here
`end`

**Comment by bradleyjucsc** [comment]
One comment, but otherwise lgtm. Build's broken though.

**Comment by fsproru** [comment]
@dvillarama, try to move `gem 'debugger'` to development group **only** in Gemfile. It should fix the error on Travis CI

**Comment by bobcook** [comment]
grab

I have tried for weeks to install Zestphone.  I have never been past this screen.  I also can not make it run at all on Heroku.  Can you offer me suggestions?

**Comment by bradleyjucsc** [comment]
Bob,

Tl;dr: I _think_ what's going on is that you need to add this to your url:
"?agent-number=<your phone number>".

The way zestphone works is that you are a call center agent, and when you
try to make an outbound call the agent's phone rings. When you pick up, an
outbound call will be made to the number you typed into the box. Note that
you can't use the same number for the agent and the "customer".

If adding your phone number to the url doesn't fix the problem, I'd be
happy to get more in depth with you on the issue your having.

-Brad

On Jan 15, 2014, at 7:45 PM, Bob Cook notifications@github.com wrote:

[image: grab]https://f.cloud.github.com/assets/20475/1927598/5e7f0b76-7e60-11e3-86dd-e89d5b2ebb49.png

I have tried for weeks to install Zestphone. I have never been past this
screen. I also can not make it run at all on Heroku. Can you offer me
suggestions?

—
Reply to this email directly or view it on
GitHubhttps://github.com/ZestFinance/zestphone/pull/1#issuecomment-32440144
.

**Comment by bobcook** [comment]
robert.cook@hrisupport.com
